feat: Update Support Scheduler docs for the Ireland release (#447)
edgexfoundry/edgex-docs: docs_src/microservices/support/scheduler/Ch-Scheduler.md

# Scheduler

![image](EdgeX_SupportingServicesScheduling.png)

## Introduction

> !!! edgey "Edgex 2.0"

Reviewer: There isn't any content for this edgey note.

Should look something like this (note the 4 space indent):
```
!!! edgey "Edgex 2.0"
    For EdgeX 2.0 xyz is new and the REST API provided by the Support Scheduler has changed to use DTOs (Data Transfer Objects) for all responses and for all POST/PUT/PATCH requests. All query APIs (GET) which return multiple objects, such as /all, provide `offset` and `limit` query parameters. 
```

Author: Updated. There is no new API in the Support Scheduler, so only removed 'xyz is new'.

# Scheduler

![image](EdgeX_SupportingServicesScheduling.png)

> ## Introduction

Reviewer: Please add an `!!! edgey` note that states what is new in the Scheduler for 2.0. One would be the revamped API using DTOs. Here is an example of an  `!!! edgey` note: 
https://github.com/edgexfoundry/edgex-docs/blob/master/docs_src/microservices/application/Triggers.md
Renders like this:
![image](https://user-images.githubusercontent.com/44779287/124190851-cb32ce80-da77-11eb-95f8-68e5e6ed2e0e.png)

Author: Updated.

> The support scheduler micro service provide an internal EdgeX “clock” that can kick off operations in any EdgeX service.  At a configuration specified time (called an **interval**), the service calls on any EdgeX service API URL via REST to trigger an operation (called an **interval action**).  For example, the scheduler service periodically calls on core data APIs to clean up old sensed events that have been successfully exported out of EdgeX.

Reviewer: ```suggestion
The support scheduler microservice provides an internal EdgeX “clock” that can kick off operations in any EdgeX service.  At a configuration specified time (called an **interval**), the service calls on any EdgeX service API URL via REST to trigger an operation (called an **interval action**).  For example, the scheduler service periodically calls on core data APIs to clean up old sensed events that have been successfully exported out of EdgeX.
```

Author: Updated.

### Default Interval Actions

Scheduled interval actions configured by default with the reference implementation of the service include:

> - Clean up Core-data events/readings that have been moved ("pushed") to external applications/systems like a cloud provider. This is the "Scrubbed Pushed" operation. Scheduler parameters around this operation determine how often and where to call into core data to invoke this clean up of unneeded data operation.

Reviewer: Remove this as mark as pushed was removed.

Author: Updated.

- Clean up of Core-data events/readings that have been persisted for an extended period. In order to prevent the edge node from running out of space, these old events/readings are removed. This is the "ScrubAged" operation. Scheduler parameters around this operation determine how often and where to call into Core-data to invoke this operation to expunge of old data.

---
**NOTE**

>     The removal of both exported records and stale records occurs on a configurable schedule. By default, both of the default actions above are invoked once a day at midnight.

Reviewer: ```suggestion
    The removal of stale records occurs on a configurable schedule. By default, the default action above is invoked once a day at midnight.
```

Author: Updated.

---

### Scheduler Persistence

> Support scheduler uses a data store to persist the Interval(s) and IntervalAction(s). Persistence is accomplished the Scheduler DB located

Reviewer: ```suggestion
Support scheduler uses a data store to persist the Interval(s) and IntervalAction(s). Persistence is accomplished by the Scheduler DB located
```

Author: Updated.

Default intervals for use with default interval actions
|Property|Default Value|Description|
|---|---|---|
|Name|midnight|Name of the every day at midnight interval|
|Start|20180101T000000|Indicates the start time for the midnight interval which is a midnight, Jan 1, 2018 which effectively sets the start time as of right now since this is in the past|
|Interval|24h|defines a frequency of every 24 hours|
### IntervalActions.IntervalActions.ScrubAged
Configuration of the core data clean old events operation which is to kick off every midnight
|Property|Default Value|Description|
|---|---|---|
|Name|scrub-aged-events|name of the interval action|
|Host|localhost|run the request on core data assumed to be on the localhost|
|Port|59880|run the request against the default core data port|
|Protocol|http|Make a RESTful request to core data|
|Method|DELETE|Make a RESTful delete operation request to core data|
|Target|core-data|target core data|
|Path|/api/v2/event/age/604800000000000|request core data's remove old events API with parameter of 7 days |
|Interval|midnight|run the operation every midnight as specified by the configuration defined interval|

## API Reference

> [Support Scheduler API Reference](../../../api/supporting/Ch-APISupportingServicesScheduler.md)

Reviewer: Good place for an `!!! edgey` note about the API changes for 2.0

Author: PR #448 is noted for the change. Should adding the same note here?
![image](https://user-images.githubusercontent.com/41656797/124208277-cb83a600-db19-11eb-96cf-9f525dc1a8cc.png)
